### PR TITLE
Add tests and implementation for custom create and deploy messages

### PR DESCRIPTION
**Pull request by therealplato**
This PR adds fields to playbook yml:

```
messages:
  created: a go {{.Template}}
  deployed: another {{ .Template }}
```

If these are invalid templates, the playbook will fail to validate.
If the fields are missing, the current default message is shown.
If the fields are present, both the current default and the custom message are shown.

Attachments for create and deploy now have green bars in Slack.

### CODE REVIEW

File: deployment/playbook_test.go [hunk 1]
```diff
@@ -22,6 +22,9 @@ vars:
   - version
   - assets_version
   - owner
+messages:
+  created: "help I'm stuck in a test {{ .ID }}"
+  deployed: "help I'm still stuck in a test {{ .ID }}"
```
Comment (macat) [inline deployment/playbook_test.go]: 😃

File: services/deployment.go [hunk 2]
```diff
@@ -1,8 +1,10 @@
 package services
 
 import (
+	"bytes"
 	"errors"
 	"fmt"
+	"html/template"
```
Comment (therealplato) [inline services/deployment.go]: `text/template` ... goimports likes to change this on me

File: deployment/playbook.go [hunk 3]
```diff
@@ -60,6 +62,12 @@ func (p *Playbook) Validate() error {
 	if len(p.Tasks) == 0 {
 		return errors.New("Playbook requires at least 1 task")
 	}
+	for key, value := range p.Messages {
+		_, err := template.New(key).Parse(value)
+		if err != nil {
+			return fmt.Errorf("Playbook had an invalid message template: \"%s\"", value)
```
Comment (bobbytables) [inline deployment/playbook.go]: One thing that I love that the go stdlib's do is that they include the package name in the error:

```go
fmt.Errorf("broadway/playbook: ...")
```

What do you think about maybe doing that here?

**Comment by bobbytables** [comment]
Other than that optional comment I think this looks great. 👍

**Comment by cored** [comment]
+1

File: services/deployment.go (hunk 2 above)
Comment (macat) [inline services/deployment.go]: can you update this?

**Pull request merged**